Lib/test/test_bool.py

```diff
@@ -58,8 +58,16 @@ def test_math(self):
         self.assertEqual(-True, -1)
         self.assertEqual(abs(True), 1)
         self.assertIsNot(abs(True), True)
-        self.assertEqual(~False, -1)
-        self.assertEqual(~True, -2)
+        with self.assertWarns(DeprecationWarning):
+            # We need to put the bool in a variable, because the constant
+            # ~False is evaluated at compile time due to constant folding;
```

Comment:
We should test this behavior separately (doing something like `with assertWarns(DeprecationWarning): exec("~False")`).

Reply:
Do you mean instead of or in addition to the current test? We originally had `eval(~True)` but I figured it's clearer to have the warning context only around the operation and not around a comparably complex `eval()` or `exec()`. I'm happy to change if there's a benefit of these though.

Reply:
We should have both. The existing tests check that the deprecation warning is emitted correctly at runtime. The new tests would ensure that if the operation occurs at compile time, we still emit the DeprecationWarning.